sccameraview/src/main/java/com/example/sccameraview/Camera2View.java

```diff
@@ -369,6 +379,70 @@ public int compare(Size lhs, Size rhs) {
         }
     }
 
+    @Override
+    public void takePicture() {
+        imageReader = ImageReader.newInstance(previewSize.getWidth(), previewSize.getHeight(), ImageFormat.JPEG, 1);
+        imageReader.setOnImageAvailableListener(mImageAvailableListener, backgroundHandler);
+
+        try {
+            final int sensorOrientation = cameraManager.getCameraCharacteristics(cameraIdString).get(
+                    CameraCharacteristics.SENSOR_ORIENTATION);
+            cameraDevice.createCaptureSession(Collections.singletonList(imageReader.getSurface()),
+                new CameraCaptureSession.StateCallback() {
+                    @Override
+                    public void onConfigured(@NonNull CameraCaptureSession session) {
+                        try {
+                        CaptureRequest.Builder captureRequestBuilder = cameraDevice.createCaptureRequest(
+                                CameraDevice.TEMPLATE_STILL_CAPTURE);
+                        captureRequestBuilder.addTarget(imageReader.getSurface());
+                        captureRequestBuilder.set(CaptureRequest.CONTROL_AF_MODE,
+                                CaptureRequest.CONTROL_AF_MODE_OFF);
+                        captureRequestBuilder.set(CaptureRequest.CONTROL_AE_MODE,
+                                CaptureRequest.CONTROL_AE_MODE_ON);
+                        captureRequestBuilder.set(CaptureRequest.FLASH_MODE,
+                                CaptureRequest.FLASH_MODE_OFF);
+                        captureRequestBuilder.set(CaptureRequest.JPEG_ORIENTATION,
+                                (sensorOrientation + getLayoutDirection() * (frontFacingCameraActive ? 1 : -1) +
+                                        360) % 360);
+
+                        session.stopRepeating();
+                        session.capture(captureRequestBuilder.build(),
+                            new CameraCaptureSession.CaptureCallback() {
+                                @Override
+                                public void onCaptureCompleted(@NonNull CameraCaptureSession session,
+                                                               @NonNull CaptureRequest request,
+                                                               @NonNull TotalCaptureResult result) {
+                                    startPreview();
```

Comment:
I do not think we want to automatically start the preview. Instead, we should create a callback, which will notify the client if set.